### DISCUSSION TITLE
How are you implementing Holes in the Terrain 3D 0.9 beta?

### DISCUSSION THREAD

**Discussion by Chevifier** [TokisanGames/Terrain3D#279] [Questions] [answered]
I'm super interested in seeing how you added punching holes in the terrain from a learning standpoint. Would be nice if you explained it in you Part 2 video. Like a high level explanation. I want to try and do it in procedural way. Ill definitely be waiting for you part 2 upload. I could just look at the source but what fun is that. lol

**Comment by TokisanGames** [comment, marked as answer]
The video is for usage, not how it works, otherwise it would be 4-8 hours.

Holes in Physics
https://github.com/TokisanGames/Terrain3D/blob/65f36b1d11e70e42e10c5b97c33ee4251c5df260/src/terrain_3d.cpp#L297-L300
https://github.com/TokisanGames/Terrain3D/blob/65f36b1d11e70e42e10c5b97c33ee4251c5df260/src/terrain_3d.cpp#L340

Holes in mesh
https://github.com/TokisanGames/Terrain3D/blob/65f36b1d11e70e42e10c5b97c33ee4251c5df260/src/shaders/main.glsl#L119-L121

**Reply by Chevifier** [reply]
Does __FLT_MAX__ essentially breaks the collision at that area i.e no collision?
I'm still fairly new to C++ been learning it to make my own extensions. Already made a few simple ones.

**Reply by TokisanGames** [reply]
Godot and Jolt physics respond to their respective constants by putting a hole in collision. I believe it is a supported feature, not breaking it.

**Reply by TokisanGames** [reply]
With the release of the second tutorial videos, I demonstrated how to use holes.
